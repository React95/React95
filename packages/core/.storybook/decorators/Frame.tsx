import React from 'react';
import '../../components/GlobalStyle/GlobalStyle.css';
import * as themes from '../../components/themes/all';

const Frame = (Story, { globals }) => {
  const { selectedTheme } = globals;

  const theme = themes[selectedTheme] || themes.win95;

  React.useEffect(() => {
    document.body.classList.add(theme);

    return () => {
      document.body.classList.remove(theme);
    };
  }, [theme]);

  return (
    <div style={{ padding: 10 }}>
      <Story />
    </div>
  );
};

export default Frame;
