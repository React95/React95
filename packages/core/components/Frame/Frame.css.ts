import { createRainbowSprinkles, defineProperties } from 'rainbow-sprinkles';
import {
  background,
  borderRadius,
  borders,
  colors,
  displayAndBoxModel,
  font,
  outline,
  positioning,
  shadows,
  text,
  zIndices,
} from './props';
import type { SprinklesFn } from 'rainbow-sprinkles/createRuntimeFn';

const displayAndBoxModelAtomicProps = defineProperties({
  dynamicProperties: displayAndBoxModel,
  shorthands: {
    size: ['height', 'width'],
    h: ['height'],
    w: ['width'],
    minH: ['minHeight'],
    minW: ['minWidth'],

    m: ['margin'],
    mr: ['marginRight'],
    ml: ['marginLeft'],
    mt: ['marginTop'],
    mb: ['marginBottom'],
    marginX: ['marginLeft', 'marginRight'],
    marginY: ['marginTop', 'marginBottom'],
    mx: ['marginLeft', 'marginRight'],
    my: ['marginTop', 'marginBottom'],

    p: ['padding'],
    pr: ['paddingRight'],
    pl: ['paddingLeft'],
    pt: ['paddingTop'],
    pb: ['paddingBottom'],
    paddingX: ['paddingLeft', 'paddingRight'],
    paddingY: ['paddingTop', 'paddingBottom'],
    px: ['paddingLeft', 'paddingRight'],
    py: ['paddingTop', 'paddingBottom'],
  },
});

const positioningAtomicProps = defineProperties({
  dynamicProperties: positioning,
});

const colorsAtomicProps = defineProperties({
  dynamicProperties: colors,
  shorthands: {
    bgColor: ['backgroundColor'],
  },
});

const backgroundAtomicProps = defineProperties({
  dynamicProperties: background,
  shorthands: {
    bg: ['background'],
  },
});

const bordersAtomicProps = defineProperties({
  dynamicProperties: borders,
});

const borderRadiusAtomicProps = defineProperties({
  dynamicProperties: borderRadius,
});

const shadowsAtomicProps = defineProperties({
  dynamicProperties: shadows,
});

const outlineAtomicProps = defineProperties({
  dynamicProperties: outline,
});

const fontAtomicProps = defineProperties({
  dynamicProperties: font,
});

const textAtomicProps = defineProperties({
  dynamicProperties: text,
});

const zIndicesAtomicProps = defineProperties({
  dynamicProperties: zIndices,
});

export const sprinkles: SprinklesFn<
  [
    typeof colorsAtomicProps,
    typeof shadowsAtomicProps,
    typeof zIndicesAtomicProps,
    typeof displayAndBoxModelAtomicProps,
    typeof positioningAtomicProps,
    typeof backgroundAtomicProps,
    typeof bordersAtomicProps,
    typeof borderRadiusAtomicProps,
    typeof outlineAtomicProps,
    typeof fontAtomicProps,
    typeof textAtomicProps,
  ]
> = createRainbowSprinkles(
  colorsAtomicProps,
  shadowsAtomicProps,
  zIndicesAtomicProps,
  displayAndBoxModelAtomicProps,
  positioningAtomicProps,
  backgroundAtomicProps,
  bordersAtomicProps,
  borderRadiusAtomicProps,
  outlineAtomicProps,
  fontAtomicProps,
  textAtomicProps,
);

export type Sprinkles = Parameters<typeof sprinkles>[0];
