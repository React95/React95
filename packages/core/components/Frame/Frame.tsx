import React, {
  ComponentPropsWithRef,
  ElementType,
  ForwardedRef,
  forwardRef,
} from 'react';
import { Sprinkles, sprinkles } from './sprinkles.css';
import cn from 'classnames';

type FixedForwardRef = <T, P = object>(
  render: (props: P, ref: React.Ref<T>) => React.ReactNode,
) => (props: P & React.RefAttributes<T>) => React.ReactNode;

const fixedForwardRef = forwardRef as FixedForwardRef;

type DistributiveOmit<T, TOmitted extends PropertyKey> = T extends any
  ? Omit<T, TOmitted>
  : never;

export type FrameProps<TAs extends ElementType = 'div'> = {
  as?: TAs;
} & DistributiveOmit<
  ComponentPropsWithRef<ElementType extends TAs ? 'div' : TAs>,
  'as'
> &
  Sprinkles;

const FrameComponent = <TAs extends ElementType>(
  props: FrameProps<TAs>,
  ref: ForwardedRef<any>,
) => {
  const { as, children, ...rest } = props;
  const Component = as || 'div';
  const { className, style, otherProps } = sprinkles(rest);

  return (
    <Component
      {...otherProps}
      style={{ ...style, ...otherProps.style }}
      className={cn(className, otherProps.className)}
      ref={ref}
    >
      {children}
    </Component>
  );
};

export const Frame = fixedForwardRef(FrameComponent);
