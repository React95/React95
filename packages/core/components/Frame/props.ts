import { contract } from '../themes/contract.css';

export const positioning = {
  position: true,
  top: true,
  right: true,
  bottom: true,
  left: true,
  pointerEvents: true,
} as const;

export const displayAndBoxModel = {
  display: true,
  overflow: true,
  boxSizing: true,

  grid: true,
  gridArea: true,
  gridAutoColumns: true,
  gridAutoFlow: true,
  gridAutoRows: true,
  gridColumn: true,
  gridColumnEnd: true,
  gridColumnStart: true,
  gridRow: true,
  gridRowEnd: true,
  gridRowStart: true,
  gridTemplate: true,
  gridTemplateAreas: true,
  gridTemplateColumns: true,
  gridTemplateRows: true,

  flex: true,
  flexBasis: true,
  flexDirection: true,
  flexFlow: true,
  flexGrow: true,
  flexShrink: true,
  flexWrap: true,

  justifyContent: true,
  justifyItems: true,
  justifySelf: true,
  alignContent: true,
  alignItems: true,
  alignSelf: true,

  gap: contract.space,

  width: contract.space,
  maxWidth: contract.space,
  minWidth: contract.space,
  height: contract.space,
  maxHeight: contract.space,
  minHeight: contract.space,

  margin: contract.space,
  marginBlock: contract.space,
  marginBlockEnd: contract.space,
  marginBlockStart: contract.space,
  marginBottom: contract.space,
  marginInline: contract.space,
  marginInlineEnd: contract.space,
  marginInlineStart: contract.space,
  marginLeft: contract.space,
  marginRight: contract.space,
  marginTop: contract.space,

  padding: contract.space,
  paddingBlock: contract.space,
  paddingBlockEnd: contract.space,
  paddingBlockStart: contract.space,
  paddingBottom: contract.space,
  paddingInline: contract.space,
  paddingInlineEnd: contract.space,
  paddingInlineStart: contract.space,
  paddingLeft: contract.space,
  paddingRight: contract.space,
  paddingTop: contract.space,
} as const;

export const colors = {
  backgroundColor: contract.colors,
  color: contract.colors,
} as const;

export const background = {
  background: true,
  backgroundAttachment: true,
  backgroundImage: true,
  backgroundPosition: true,
  backgroundRepeat: true,
  backgroundSize: true,
  backgroundOrigin: true,
  backgroundClip: true,
} as const;

export const borders = {
  border: true,
  borderStyle: true,
  borderWidth: true,
  borderColor: contract.colors,
  borderTop: true,
  borderTopColor: contract.colors,
  borderTopStyle: true,
  borderTopWidth: true,
  borderRight: true,
  borderRightColor: contract.colors,
  borderRightStyle: true,
  borderRightWidth: true,
  borderBottom: true,
  borderBottomColor: contract.colors,
  borderBottomStyle: true,
  borderBottomWidth: true,
  borderLeft: true,
  borderLeftColor: contract.colors,
  borderLeftStyle: true,
  borderLeftWidth: true,
} as const;

export const borderRadius = {
  borderBottomLeftRadius: true,
  borderBottomRightRadius: true,
  borderImage: true,
  borderImageOutset: true,
  borderImageRepeat: true,
  borderImageSlice: true,
  borderImageSource: true,
  borderImageWidth: true,
  borderRadius: true,
  borderTopLeftRadius: true,
  borderTopRightRadius: true,
} as const;

export const shadows = {
  boxShadow: contract.shadows,
} as const;

export const outline = {
  outline: true,
  outlineColor: true,
  outlineStyle: true,
  outlineWidth: true,
} as const;

export const font = {
  font: true,
  fontFamily: true,
  fontStyle: true,
  fontSize: true,
  fontWeight: true,
  fontVariant: true,
  fontSizeAdjust: true,
} as const;

export const text = {
  color: contract.colors,
  direction: true,
  letterSpacing: true,
  lineHeight: true,
  textAlign: true,
  textDecoration: true,
  textIndent: true,
  textShadow: true,
  textTransform: true,
  verticalAlign: true,
  whiteSpace: true,
  wordSpacing: true,
  wordBreak: true,
} as const;

export const zIndices = {
  zIndex: contract.zIndices,
} as const;
