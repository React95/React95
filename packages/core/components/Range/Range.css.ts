import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';
import { calc } from '@vanilla-extract/css-utils';

const track = {
  width: '100%',
  height: contract.space[4],
  backgroundColor: contract.colors.borderDarkest,
  cursor: 'pointer',
  borderStyle: 'solid',
  borderTopWidth: contract.space[1],
  borderTopColor: contract.colors.borderLight,
  borderRightWidth: contract.space[2],
  borderRightColor: contract.colors.borderLightest,
  borderBottomWidth: contract.space[1],
  borderBottomColor: contract.colors.borderLighter,
  borderLeftWidth: contract.space[2],
  borderLeftColor: contract.colors.borderLightest,
  boxShadow: `0px 1px 0px 0px ${contract.colors.borderLightest}`,
} as const;

const thumb = {
  width: contract.space[12],
  height: contract.space[20],
  backgroundColor: contract.colors.material,
  cursor: 'pointer',
  marginTop: calc.negate(contract.space[8]),
  borderStyle: 'solid',
  borderTopWidth: contract.space[1],
  borderTopColor: contract.colors.borderLightest,
  borderRightWidth: contract.space[1],
  borderRightColor: contract.colors.borderDarkest,
  borderBottomWidth: contract.space[1],
  borderBottomColor: contract.colors.borderDarkest,
  borderLeftWidth: contract.space[1],
  borderLeftColor: contract.colors.borderLightest,
  boxShadow: `inset 0px -1px 0px ${contract.colors.borderDark},
  inset -1px 0px 0px ${contract.colors.borderDark},
  inset 0px 1px 0px ${contract.colors.borderLighter},
  inset 1px 0px 0px ${contract.colors.borderLighter}`,
  appearance: 'none',
} as const;

export const range = style({
  appearance: 'none',
  width: '100%',
  selectors: {
    '&::hover,&::focus,&::active': { outline: 'none' },
    '&::-webkit-slider-runnable-track': track,
    '&::-moz-range-track': track,
    '&::-webkit-slider-thumb': thumb,
    '&::-moz-range-thumb': thumb,
    '&:focus::-webkit-slider-runnable-track': {
      backgroundColor: contract.colors.borderDarkest,
    },
  },
});
