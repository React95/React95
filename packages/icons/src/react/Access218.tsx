import React from 'react';
import { IconProps } from '../iconType';
import { BaseIcon } from '../internal';
import Access218_32x32_4 from '../../png/Access218_32x32_4.png';

export const access218Data = {
  '32x32_4': {
    imageSrc: Access218_32x32_4 as string,
    width: 32,
    height: 32,
  },
};

export interface Access218Props extends IconProps {
  /**
   * Icon variant to use.
   * also provides default styling with the correct height and width
   **/
  variant?: '32x32_4';
}

export const Access218: React.FC<Access218Props> = ({
  variant = '32x32_4',
  ...rest
}) => {
  const image = access218Data[variant];

  return (
    <BaseIcon
      width={image.width}
      height={image.height}
      src={image.imageSrc}
      {...rest}
    />
  );
};
