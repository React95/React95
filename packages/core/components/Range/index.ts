import { Range } from './Range';

export default Range;
