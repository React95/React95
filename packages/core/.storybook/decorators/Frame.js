import React from 'react';
import { GlobalStyle } from '../../components';

const Frame = Story => (
  <div style={{ padding: 10 }}>
    <GlobalStyle />
    <Story />
  </div>
);

export default Frame;
