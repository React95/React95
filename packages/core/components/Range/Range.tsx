import * as React from 'react';
import { Frame, FrameProps } from '../Frame/Frame';
import { range } from './Range.css';
import cn from 'classnames';

export type RangeProps = Omit<FrameProps<'input'>, 'as'>;

export const Range = React.forwardRef<HTMLInputElement, RangeProps>(
  (rest, ref) => (
    <Frame
      {...rest}
      ref={ref}
      className={cn(range, rest.className)}
      as="input"
      type="range"
    />
  ),
);
