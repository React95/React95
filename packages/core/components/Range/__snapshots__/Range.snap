// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Range /> > Snapshot > should match snapshot 1`] = `
<div>
  <input
    class="Range_range__1i4w9z20"
    type="range"
  />
</div>
`;
